### PR TITLE
Make String.{Try}CopyTo public

### LINKED ISSUE #51061: API Proposal: String.{Try}CopyTo(Span<char>)
## Background and Motivation

When working with text (strings and spans), it's relatively common to need to copy a string to a span, which can be done with:
```C#
string.AsSpan().CopyTo(destination)
```
We can make this simpler and faster by adding such a CopyTo directly to string:
```C#
string.CopyTo(destination)
```
Same with TryCopyTo.  For example, Number.Formatting.cs in corelib has this:
https://github.com/dotnet/runtime/blob/f0f3b845c4abaa726b5e30ae0ad7c7a689678946/src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs#L757-L769
by changing the `source.AsSpan().TryCopyTo(destination)` there to `source.TryCopyTo(destination)` (with an implementation of TryCopyTo on string that's the same 7-line implementation on span), we get this diff:
image

## Proposed API

```diff
namespace System
{
    public sealed class String
    {
        ...
        public void CopyTo(int sourceIndex, char[] destination, int destinationIndex, int count);
+       public void CopyTo(Span<char> destination);
+       public bool TryCopyTo(Span<char> destination);
    }
}

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes #51061

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs [hunk 1]
```diff
@@ -168,9 +168,9 @@ static ReadOnlySpan<char> WritePostEB(ReadOnlySpan<char> label, Span<char> desti
             {
                 int size = PostEBPrefix.Length + label.Length + Ending.Length;
                 Debug.Assert(destination.Length >= size);
-                PostEBPrefix.AsSpan().CopyTo(destination);
+                PostEBPrefix.CopyTo(destination);
                 label.CopyTo(destination.Slice(PostEBPrefix.Length));
-                Ending.AsSpan().CopyTo(destination.Slice(PostEBPrefix.Length + label.Length));
+                Ending.CopyTo(destination.Slice(PostEBPrefix.Length + label.Length));
```
Comment (GrabYourPitchforks) [inline src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs:173]: Looks like this is a netcoreapp2.1 assembly? Surprised CI isn't complaining.

Comment (stephentoub) [reply src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs:173]: I don't think so?
https://github.com/dotnet/runtime/blob/edce31f285be386c952a8f538be1b5e3642fe5c1/src/libraries/System.Security.Cryptography.Encoding/src/System.Security.Cryptography.Encoding.csproj#L4

Comment (GrabYourPitchforks) [reply src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs:173]: Well, then it looks like I should stop paying attention to source.dot.net. :)

https://source.dot.net/System.Security.Cryptography.Encoding/System.Security.Cryptography.Encoding.csproj.html

```xml
<TargetFramework>netcoreapp2.1</TargetFramework>
```

**Review by GrabYourPitchforks — APPROVED** [review]

File: src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs:173]: @alexperovich, where is that file coming from?  I can't find anything in the repo or in the history of the csproj that looks like that?

**Pull request merged**